zrml/swaps/rpc/src/lib.rs

```diff
@@ -67,18 +76,36 @@ impl From<Error> for i64 {
     }
 }
 
+macro_rules! get_spot_price_rslt {
+    (
+        $api_ref:expr,
+        $asset_in:expr,
+        $asset_out:expr,
+        $at:expr,
+        $pool_id:expr
+    ) => {
+        $api_ref
+            .get_spot_price($at, $pool_id, $asset_in, $asset_out)
+            .map_err(|e| RpcError {
+                code: ErrorCode::ServerError(Error::RuntimeError.into()),
+                message: "Unable to get spot price.".into(),
+                data: Some(format!("{:?}", e).into()),
+            })
+    };
+}
```

Comment:
Didn't use a function here because it would require a lot of bound definitions just like `impl SwapsApi for Swaps`